frontend/lib/resolution/VarScopeVisitor.cpp

```diff
@@ -275,6 +303,29 @@ void VarScopeVisitor::exitScope(const AstNode* ast, RV& rv) {
           }
         }
       }
+    } else if (auto s = ast->toSelect()) {
+      handleSelect(s, rv);
+      if (parentFrame != nullptr) {
+        //if no otherwise block, returnsOrThrows only if all present blocks 
+        // returnsOrThrows and one of them is paramTrue
+        //if otherwise block, returnsOrThrows only if all present blocks 
+        // returnsOrThrows
```

Comment:
I'm seeing your point. I think it hinges on the assumptions that if there's any paramTrue (or paramFalse) condition, all of the conditions are either paramTrue or paramFalse. 

I started with this assumption, and I'm now realizing it may still be present elsewhere in the code, but I don't think its true:
```
proc foo(x: int) {
  select 2 {
    when x do writeln("not returning");
    when 2 do return;
  }
}
```
Here, the `when 2` condition is paramTrue, but the `when x` condition is neither paramTrue or paramFalse. The code above would mark the `select` as always returning, but it doesn't return when `x=2`. 

You're right that my formulation is too strict. I think it needs to be:
- if at least one frame is paramTrue, returnsOrThrows if all frames preceding and including the first paramTrue frame return or throw
- if no frames are paramTrue, then returnsOrThrows if there is an otherwise block and all frames return or throw.